core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -94,7 +98,16 @@ private[spark] class DiskBlockManager(
       } else {
         val newDir = new File(localDirs(dirId), "%02x".format(subDirId))
         if (!newDir.exists()) {
-          Files.createDirectory(newDir.toPath)
+          val path = newDir.toPath
+          Files.createDirectory(path)
+          if (shuffleServiceRemoveShuffleEnabled) {
```

Comment:
We can't do it here because we need to make sure all blockmgr dirs are created with group write permission, not matter what type of file causes the dir creation. We potentially can in `createTempFileWith`